stores/directory/local.py

```diff
@@ -23,42 +26,81 @@ def __init__(self, local_dir: Union[str, Path]):
         assert local_dir_path.is_dir(), f"Given path {local_dir_path} is not a directory."
 
         # Store that location against the class
-
         self.local_path = local_dir_path
 
-
     def add_file(self, file: Path):
         # Abstract methods can remain without implementation for now.
         pass
 
     def has_lone_file_matching(self, pattern: str) -> bool:
-        #  To be implemented in local directory store 1 #5 
-        ...
-
-    def save_lone_file_matching(self, pattern: str, save_as: str) -> Path:
-        #  To be implemented in local directory store 1 #5 
-        ...
+        # Grab a list of files matching the regex pattern to determine how many exist.
+        matching_files = [
+            f for f in self.get_file_names() if re.search(pattern, f)]
+
+        if len(matching_files) == 1:  # 1 file matched
+            return True
+        elif len(matching_files) == 0:  # 0 file matched
+            raise FileNotFoundError(f"No files found matching given pattern '{pattern}' in directory {self.local_path}")          
+        elif len(matching_files) > 1:  # 2+ files matched
+            raise FileNotFoundError(f"More than 1 file found that matches the regex pattern '{pattern}' in directory {self.local_path}")
```

Comment:
can you stick `matching_files` in this error message please, will make it easy to debug the issue.